core/src/main/java/org/apache/accumulo/core/client/impl/TableOperationsImpl.java

```diff
@@ -1208,11 +1207,20 @@ public void online(String tableName) throws AccumuloSecurityException, AccumuloE
   @Override
   public void online(String tableName, boolean wait) throws AccumuloSecurityException, AccumuloException, TableNotFoundException {
     checkArgument(tableName != null, "tableName is null");
+
+    long cacheResetCount = Tables.getCacheResetCount();
+
     String tableId = Tables.getTableId(context.getInstance(), tableName);
 
     /**
      * ACCUMULO-4574 if table is already online return without executing fate operation.
      */
+
+    // getTableId may have side effect of clearing zookeeper cache - no need to clear again.
+    if (cacheResetCount == Tables.getCacheResetCount()) {
+      Tables.clearCacheByPath(context.getInstance(), Constants.ZTABLES + "/" + tableId + Constants.ZTABLE_STATE);
```

Comment:
I suspect other methods in Tables class would benefit from having a boolean that indicates if cache should be used.  If you go with the boolean option, could possibly have a private method ` Tables.clearCachedTableState(String tableId)` that's used in the implementation.